drafts/draft-ietf-ippm-ioam-data-integrity.xml

```diff
@@ -654,45 +654,91 @@
+      The method MUST use AES-GMAC (<xref target="AES"/>
+      <xref target="NIST.800-38D"/>), a block
       cipher mode of operation providing data origin authentication, which is
       also a specialization of the Galois/Counter Mode (GCM). The GCM
       authenticated encryption operation has four inputs: a secret key, an
       Initialization Vector (IV), a plaintext, and Additional Authenticated Data
       (AAD). It has two outputs: a ciphertext whose length is identical to the
       plaintext and an Authentication Tag. GMAC is the special case of GCM in
       which the plaintext has a length of zero. Therefore, the empty ciphertext
-      output is ignored, and the only output is the Authentication Tag.</t>
+      output is ignored, and the only output is the Authentication Tag.
+      GMAC supports all AES key sizes (128, 192, 256 bits).</t>
 
-      <t>In this method, the AES-GMAC Authentication Tag MUST NOT be truncated,
+      <t>In this method, the GMAC Authentication Tag MUST NOT be truncated,
       meaning its size MUST always be 16 octets (i.e., a full Authentication
-      Tag). More security
-      recommendations are discussed in <xref target="Security"/>. Below, we
-      refer to the AES-GMAC Initialization Vector (IV) as the nonce, and to the
-      AES-GMAC Authentication Tag as the Integrity Check Value (ICV).</t>
+      Tag). Below, we
+      refer to the GMAC Initialization Vector (IV) as the nonce, and to the
+      GMAC Authentication Tag as the Integrity Check Value (ICV).</t>
 
       <section anchor="IPM-Encap" title="Encapsulating node">
-        <t>The encapsulating node generates a nonce and stores it in the Nonce
-        field of the IOAM Integrity Protection header (the Nonce length
-        field is set accordingly). It MUST be a 12-octet
-        nonce, based on the "Deterministic Construction" (see
-        <xref target="Security"/>). The Method ID field MUST be set to 1, as
-        defined in <xref target="IOAM-IP-registry"/>.</t>
+        <t>The encapsulating node generates a nonce based on some mandatory
+        requirements (see <xref target="NonceRequirements"/>) and stores it in
+        the Nonce field of the IOAM Integrity Protection header (the Nonce
+        Length field is set accordingly). The Method-ID field MUST be set to 1,
+        as defined in <xref target="IOAM-IP-registry"/>.</t>
 
         <t>The Integrity Check Value (ICV) is the result of a GMAC operation
         over a selection of header fields (see
         <xref target="SelectHdrFields"/>) and immutable IOAM-Data-Fields added
-        by the encapsulating node. With the nonce provided to
-        GMAC, the encapsulating node performs the following operation:
+        by the encapsulating node. In the case of pre-allocated
+        IOAM-Data-Fields, only those updated by the encapsulating node are part
+        of the GMAC operation. With the nonce provided to GMAC, the
```

Comment:
I agree, although I tried to keep it a bit more generic in case someone defines a new Option-Type that mimics and resembles the (pre-allocated) Trace Option-Type. On the other hand, should it be the case, I guess it would be fine to add/update some text to apply the same behavior on such new Option-Type for the integrity protection.

Based on your suggestion, how about:

_In the case the Pre-allocated Trace Integrity Protected Option-Type is used, the encapsulating node only includes the IOAM-Data-Fields that correspond to the itself, i.e., "node data list [n]" (see RFC 9197) into the GMAC operation._

... or we could try to keep it generic. We could reuse RFC 9197's wording, i.e., "_a container of node data fields with pre-allocated space_". Thoughts?

Reply:
Let's be specific. It is hard to solve potential issues for the future - if we don't know the future.

Reply:
And yes, I agree with your proposed wording above - to scope things to "pre-allocated Trace Integrity Protected Option-Type"